drivers/SmartThings/zigbee-button/src/zigbee-multi-button/robb/init.lua

```diff
@@ -0,0 +1,226 @@
+  -- Emit events
+  if comp ~= nil then
+    device:emit_component_event(comp, event)
+    device:emit_event(event)
+  end
+end
+
+local do_configuration = function(driver, device)
+  -- Get the right number of endpoints
+  local endpoints = WIRELESS_REMOTE_FINGERPRINTS[device:get_model()].endpoints
+
+  device:send(device_management.build_bind_request(device, PowerConfiguration.ID, driver.environment_info.hub_zigbee_eui), 1)
+  device:send(PowerConfiguration.attributes.BatteryPercentageRemaining:configure_reporting(device, 30, 21600, 1))
+
+  for endpoint = 1, endpoints do
+    device:send(device_management.build_bind_request(device, Level.ID, driver.environment_info.hub_zigbee_eui, endpoint))
+    device:send(device_management.build_bind_request(device, OnOff.ID, driver.environment_info.hub_zigbee_eui, endpoint))
+  end
+
+  device:send(OnOff.attributes.OnOff:configure_reporting(device, 0, 600, 1))
+
+  device:send(Basic.attributes.DeviceEnabled:write(device, true))
+  if not device:get_field('is_group_configured') then
+    -- Configure adding hub to group once
+    driver:add_hub_to_zigbee_group(0xE902)
+    -- Add two more groups if it is the 8x wall switch
+    if endpoints == 4 then
+      driver:add_hub_to_zigbee_group(0xE903)
+      driver:add_hub_to_zigbee_group(0xE904)
+    end
+
+    device:set_field('is_group_configured', true, { persist = true })
+  end
+end
+
+local function added_handler(self, device)
+  local supported_button_values
+  local number_of_buttons
+  local comp_id
+
+  for _, comp in pairs(device.profile.components) do
+    comp_id = comp.id
+
+    -- Special handeling of ON buttons (left column)
+    if comp_id == "button1" or comp_id == "button3" or comp_id == "button5" or comp_id == "button7" then
+      supported_button_values = { "pushed", "up_hold" }
+      number_of_buttons = 1
+    -- Special handling of OFF buttons (right column)
+    elseif comp_id == "button2" or comp_id == "button4" or comp_id == "button6" or comp_id == "button8" then
+      supported_button_values = { "pushed", "down_hold" }
+      number_of_buttons = 1
+    -- Handling of main button capability
+    elseif comp_id == "main" then
+      supported_button_values = { "pushed", "up_hold", "down_hold" }
+      number_of_buttons = WIRELESS_REMOTE_FINGERPRINTS[device:get_model()].buttons
+    end
+
+    device:emit_component_event(comp,
+      capabilities.button.supportedButtonValues(supported_button_values, { visibility = { displayed = false } }))
+    device:emit_component_event(comp,
+      capabilities.button.numberOfButtons({ value = number_of_buttons }, { visibility = { displayed = false } }))
+  end
+  device:emit_event(capabilities.button.button.pushed({ state_change = false }))
+  device:send(PowerConfiguration.attributes.BatteryPercentageRemaining:read(device))
+end
+
+local battery_perc_attr_handler = function(driver, device, value, zb_rx)
+  device:emit_event(capabilities.battery.battery(utils.clamp_value(value.value, 0, 100)))
```

Comment:
the zigbee spec says this attribute should be reported in half percents, i.e. 200 for 100% battery. Are you reporting whole percents? just curious because you're clamping to 100, implying you're getting values higher than 100

Reply:
Yes, you're right. The spec says 0 to 200 in 0.5 % - sorry for that. I just wonder if it is the same case [here](https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/blob/main/drivers/SmartThings/zigbee-button/src/zigbee-multi-button/ikea/init.lua#L108) and [here](https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/blob/main/drivers/SmartThings/zigbee-dimmer-remote/src/zigbee-battery-accessory-dimmer/IKEAofSweden/init.lua#L99) ?